### DISCUSSION TITLE
triming trailing blank lines in python files

### DISCUSSION THREAD

**Discussion by ghost** [NvChad/NvChad#2551] [Q&A] [answered]
Hello everybody
I'm from Venezuela
I`m just starting in NeoVim and using NVChad workspace, but I really don`t know how to make a function that remove any blank line at the end of any file and make an autocomand for python files.
Sorry for my bad english!
En Español:
Quiero hacer una función que remueva cualquier espacio en blanco al final de una instrucción y también remueva cualquier línea en blanco al final de los archivos python (o cualquier otro archivo)
y asignar esa función a un autocomando y a una key-binding.
Gracias

**Comment by siduck** [comment, marked as answer]
```lua
local autocmd = vim.api.nvim_create_autocmd

autocmd("FileType", {
  pattern = "python" ,
  command = "%s#\\($\\n\\s*\\)\\+\\%$## | w",
})
```

https://stackoverflow.com/questions/7495932/how-can-i-trim-blank-lines-at-the-end-of-file-in-vim

**Comment by ghost** [comment]
Thans a lot!
    El miércoles, 6 de diciembre de 2023, 11:12:19 p. m. GMT-4, Sidhanth Rathod ***@***.***> escribió:  
 
 

local autocmd = vim.api.nvim_create_autocmd

autocmd("FileType", {
  pattern = "python" ,
  command = "%s#\\($\\n\\s*\\)\\+\\%$## | w",
})
